### ISSUE TITLE
Conditional email verification not working in some scenarios

### ISSUE THREAD

**Issue by mtrezza** [parse-community/parse-server#8839]
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the [latest version of Parse Server](https://github.com/parse-community/parse-server/releases). 

### Issue Description

Conditional email verification with functions for `verifyUserEmails`, `preventLoginWithUnverifiedEmail` is not working in some scenarios. That is because `verifyUserEmails` and `preventLoginWithUnverifiedEmail` can be functions since https://github.com/parse-community/parse-server/pull/8425, but the Parse Server code has not been updated in all places. For example, in some places the value for `verifyUserEmails` is still checked with `if (this.config.verifyUserEmails) { ... }` which doesn't work if the Parse Server options is set to a function.

For example, this configuration works as expected on sign up and returns error HTTP 205 `User email is not verified.`:

```
verifyUserEmails: async () => true,
preventLoginWithUnverifiedEmail: true,
preventSignupWithUnverifiedEmail: true,
```

But this configuration does not work as expected on sign up and instead returns HTTP 201 with the Parse.User object:

```
verifyUserEmails: async () => true,
preventLoginWithUnverifiedEmail: async () => true,
preventSignupWithUnverifiedEmail: true,
```

### Environment

Server
- Parse Server version: `7.0.0-alpha.2`

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.0.0-alpha.3](https://github.com/parse-community/parse-server/releases/tag/7.0.0-alpha.3)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.0.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/7.0.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.0.0](https://github.com/parse-community/parse-server/releases/tag/7.0.0)
